crates/exex/exex/src/wal/storage.rs

```diff
@@ -0,0 +1,200 @@
+use std::{
+    fs::File,
+    io::{Read, Write},
+    ops::{Bound, RangeBounds, RangeInclusive},
+    path::{Path, PathBuf},
+};
+
+use eyre::OptionExt;
+use reth_exex_types::ExExNotification;
+use reth_tracing::tracing::debug;
+
+/// The underlying WAL storage backed by a directory of files.
+///
+/// Each notification is represented by a single file that contains a MessagePack-encoded
+/// notification.
+#[derive(Debug)]
+pub(super) struct Storage {
+    /// The path to the WAL file.
+    path: PathBuf,
+    pub(super) min_id: Option<u64>,
+    pub(super) max_id: Option<u64>,
```

Comment:
why do we need this?

Reply:
accounting for lowest/highest IDs:
- lowest ID for when we want to iterate over notifications until a certain file ID, i.e. `..file_id`. We need to know where to start, because the first file can be other than 0 (if earlier files are finalized and removed).
- highest ID for when we want to create a new file or iterate over notifications starting from a certain file ID, i.e. `file_id..`. We need to know when to end.

Reply:
can't we derive all of this from the cached entries alone?

Reply:
you're right, it's much easier. Also removed the weird range adjustment thing.